streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBWindowStore.java

```diff
@@ -48,11 +49,12 @@
 
     private static final long USE_CURRENT_TIMESTAMP = -1L;
 
-    private static class Segment extends RocksDBStore<byte[], byte[]> {
+    // use the ByteBuffer wrapper for underlying rocksDB keys since they are used for hashing data structures
+    private static class Segment extends RocksDBStore<ByteBuffer, byte[]> {
```

Comment:
I don't think the performance would be any worse than ByteBuffer: it should really be faster, both because caching the hashCodes might be a win, and because a custom version can assume more about the underlying byte[].

The wrapper can be very simple, delegating to `Arrays.hashCode` and `Arrays.equals` for the important parts. If it would help to look it an example, I have one in my pet utility-library that you can see here: https://github.com/joshng/papaya/blob/master/papaya/src/main/java/com/joshng/util/collect/ByteArrayKey.java